File: src/dotnet/PloyBus.Tests/Transport/Transactions/Messages/Handlers/Error/TestTransport.cs
namespace PolyBus.Transport.Transactions.Messages.Handlers.Error;

class TestTransport : ITransport
{
    public const string DefaultDeadLetterEndpoint = "dead-letters";
    public string DeadLetterEndpoint => DefaultDeadLetterEndpoint;
    public Task Handle(Transaction transaction) => throw new NotImplementedException();
    public bool SupportsCommandMessages => true;
    public bool SupportsDelayedCommands => true;
    public bool SupportsSubscriptions => false;

    public Task Send(Transaction transaction) => Task.CompletedTask;
    public Task Subscribe(MessageInfo messageInfo) => Task.CompletedTask;
    public Task Start() => Task.CompletedTask;
    public Task Stop() => Task.CompletedTask;
}

File: src/kotlin/src/test/kotlin/polybus/transport/transactions/messages/handlers/error/TestTransport.kt
package polybus.transport.transactions.messages.handlers.error

import polybus.transport.ITransport
import polybus.transport.transactions.Transaction
import polybus.transport.transactions.messages.MessageInfoValue

class TestTransport : ITransport {
    companion object {
        const val DEFAULT_DEAD_LETTER_ENDPOINT = "dead-letters"
    }

    override val deadLetterEndpoint: String = DEFAULT_DEAD_LETTER_ENDPOINT
    override suspend fun handle(transaction: Transaction) {
        throw NotImplementedError()
    }

    override val supportsCommandMessages: Boolean = true
    override val supportsDelayedCommands: Boolean = true
    override val supportsSubscriptions: Boolean = false

    override suspend fun subscribe(messageInfo: MessageInfoValue) {}
    override suspend fun start() {}
    override suspend fun stop() {}
}
